packages/protocol/test/authorizers/Api3TokenLock.sol.js

```diff
@@ -0,0 +1,755 @@
+          ).to.equal(0);
+          expect(await api3TokenLock.airnodeToRequesterToTokenLocks(airnodeAddress, roles.requester.address)).to.equal(
+            1
+          );
+
+          const beforeBalance2 = await api3Token.balanceOf(roles.anotherSponsor.address);
+
+          await api3TokenLock.connect(roles.anotherSponsor).unlock(airnodeAddress, roles.requester.address);
+
+          const afterBalance2 = await api3Token.balanceOf(roles.anotherSponsor.address);
+          expect(afterBalance2).to.equal(beforeBalance2.add(LOCK_AMOUNT));
+          expect(
+            await api3TokenLock.sponsorUnlockTime(airnodeAddress, roles.requester.address, roles.anotherSponsor.address)
+          ).to.equal(0);
+          expect(
+            await api3TokenLock.sponsorLockAmount(airnodeAddress, roles.requester.address, roles.anotherSponsor.address)
+          ).to.equal(0);
+          expect(await api3TokenLock.airnodeToRequesterToTokenLocks(airnodeAddress, roles.requester.address)).to.equal(
+            0
+          );
+        });
+
+        it('emits event', async () => {
+          await expect(api3TokenLock.connect(roles.sponsor).unlock(airnodeAddress, roles.requester.address))
+            .to.emit(api3TokenLock, 'Unlocked')
+            .withArgs(airnodeAddress, roles.requester.address, roles.sponsor.address, LOCK_AMOUNT);
+        });
+      });
+
+      context('caller period not expired', async () => {
+        it('reverts', async () => {
+          await expect(
+            api3TokenLock.connect(roles.sponsor).unlock(airnodeAddress, roles.requester.address)
+          ).to.revertedWith(Errors.LockPeriodNotExpired);
+        });
+      });
+
+      context('already unlocked', async () => {
+        it('reverts', async () => {
+          utils.timeTravel(ONE_MINUTE_IN_SECONDS + 1000);
+          await api3TokenLock.connect(roles.sponsor).unlock(airnodeAddress, roles.requester.address);
+
+          await expect(
+            api3TokenLock.connect(roles.sponsor).unlock(airnodeAddress, roles.requester.address)
+          ).to.revertedWith('No amount locked');
+        });
+      });
+
+      context('requester address is blocked', async () => {
+        it('reverts', async () => {
+          await api3TokenLock.connect(roles.metaAdmin).blockRequester(airnodeAddress, roles.requester.address);
+          utils.timeTravel(ONE_MINUTE_IN_SECONDS + 1000);
+
+          await expect(
+            api3TokenLock.connect(roles.sponsor).unlock(airnodeAddress, roles.requester.address)
+          ).to.revertedWith(Errors.RequesterBlocked);
+        });
+      });
+
+      context('requester is unlocking on address zero', async () => {
+        it('reverts', async () => {
+          await expect(
+            api3TokenLock.connect(roles.sponsor).unlock(hre.ethers.constants.AddressZero, roles.requester.address)
+          ).to.revertedWith(Errors.ZeroAddress);
+        });
+      });
+    });
```

Comment:
Are we missing ERROR_LOCK_PERIOD_ZERO test?

Reply:
After going through the code, it isn't possible to ever reach the ERROR_LOCK_PERIOD_ZERO require statement. Consider the two cases:

> User locks tokens for himself which sets the lockamount and also sets the unlockTime which is > 0.
> User does not unlock tokens in which case the user has both the lockamount and unlockTime = 0.

so the revert condition is never reached because the lockamount require condition is always triggered first. One way to do this would be to AND them but I dont think its necessary and opted to remove it. What do you think?

Reply:
It might be redudant to check both values for 0 so I guess it's fine to remove that check